app/models/piece.rb

```diff
@@ -85,9 +85,14 @@ def horizontal_obstruction?(destination_x, _destination_y)
 
   def move_to!(destination_x, destination_y)
     piece = Piece.find_by(x: destination_x, y: destination_y)
-    if piece.color != color
-      piece.destroy
-      update(x: destination_x, y: destination_y)
+
+    if piece.present?
+      if piece.color != color
+        piece.destroy
+        update(x: destination_x, y: destination_y)
+      end
```

Comment:
If the piece is the same color, then nothing happens the xy coordinates of the current piece will not be updated